modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java

```diff
@@ -411,6 +420,30 @@ public void updatePowwowMeeting(
 		writeJSON(actionRequest, actionResponse, jsonObject);
 	}
 
+	protected void addPowwowMeetingOccurrences(long powwowMeetingId) {
+
+		_deletePowwowMeetingOccurrence(powwowMeetingId);
+
+		JSONObject zoomMeetingJSONObject =
+			PowwowServiceProviderUtil.getZoomMeetingJSONObject(powwowMeetingId);
+
+		JSONArray occurrencesJsonArray = zoomMeetingJSONObject.getJSONArray("occurrences");
+
+		PowwowMeetingOccurrence powwowMeetingOccurrence =
+			PowwowMeetingOccurrenceLocalServiceUtil.createPowwowMeetingOccurrence(StringPool.BLANK);
+		powwowMeetingOccurrence.setPowwowMeetingId(powwowMeetingId);
+
+		for (int i = 0; i < occurrencesJsonArray.length(); i++) {
+			JSONObject occurrenceJSONObject = occurrencesJsonArray.getJSONObject(i);
+
```

Comment:
let's have a service method `addPowwowMeetingOccurrence(params...)` which accepts instance's params and some of audit fields if possible